pkg/operator/controller/callbacks.go

```diff
@@ -63,6 +63,9 @@ func addReconcileCallbacks(r *ReconcileCDI) {
 	if r.haveRoutes {
 		r.reconciler.AddCallback(&appsv1.Deployment{}, reconcileRoute)
 	}
+	if r.haveVolumeDataSourceValidator {
```

Comment:
Maybe we should just care about creating and handling our VolumePopulators without checking whether the validator is installed? For routes it makes sense as they are openshift-exclusive, but if having VolumePopulators is the k8s standard maybe we should be creating them anyway.